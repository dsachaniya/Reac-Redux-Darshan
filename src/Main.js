import React, { StrictMode } from 'react';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';
import Routes from 'routes';
import configureStore from './redux';

/**
 * Main
 * @returns {Function}
 */
function Main() {
  const store = configureStore();
  return (
    <StrictMode>
      <Provider store={store}>
        <HelmetProvider>
          <Routes />
        </HelmetProvider>
      </Provider>
    </StrictMode>
  );
}

export default Main;
