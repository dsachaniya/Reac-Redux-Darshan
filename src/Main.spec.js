import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

describe('<Main />', () => {
  it('Rendering', () => {
    const wrapper = shallow(<Main />);
    // 1. Snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
