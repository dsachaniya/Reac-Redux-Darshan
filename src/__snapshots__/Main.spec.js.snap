// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Rendering 1`] = `
<StrictMode>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <HelmetProvider
      context={Object {}}
    >
      <Connect(AppRouter) />
    </HelmetProvider>
  </Provider>
</StrictMode>
`;
